pkg/namespace/client.go

```diff
@@ -78,7 +75,18 @@ func (c *Client) run(stop <-chan struct{}) error {
 
 // IsMonitoredNamespace returns a boolean indicating if the namespace is among the list of monitored namespaces
 func (c Client) IsMonitoredNamespace(namespace string) bool {
-	_, exists, _ := c.cache.GetByKey(namespace)
+	item, exists, _ := c.cache.GetByKey(namespace)
+	if exists {
+		namespace, ok := item.(*corev1.Namespace)
+		if !ok {
+			log.Error().Err(errGettingNamespace).Msgf("Failed to get %s", namespace)
+			return false
+		}
+
+		value, found := namespace.Labels[constants.OSMKubeResourceMonitorAnnotation]
+		return found && value == c.meshName
+	}
+
```

Comment:
This shouldn't be needed if the resources are already being filtered based on the label selector.